### PR TITLE
finished reviews for showpage

### PR DESCRIPTION
**Pull request by ATrbovic**
- Migration for reviews table
- Model for reviews
- RelationMappings for reviews and shows
- Reviews Seeder
- Add the review to TV Details page
- Reviews serializer

### CODE REVIEW

File: server/src/db/migrations/20210721143134_createReviews.cjs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * @typedef {import("knex")} Knex
+ */
+
+/**
+ * @param {Knex} knex
+ */
+exports.up = async (knex) => {
+  return knex.schema.createTable("reviews", (table) => {
+    table.bigIncrements("id")
+    table.text("reviewBody").notNullable()
```
Comment (nwalberts) [inline server/src/db/migrations/20210721143134_createReviews.cjs]: just call this body

File: server/src/models/Review.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+class Review extends Model {
+  static get tableName() {
+    return "reviews"
+  }
+  
+  static get jsonSchema() {
+    return {
+      type: "object",
+      required: ["reviewBody", "score", "showId"],
```
Comment (nwalberts) [inline server/src/models/Review.js]: update reviewBody after update to migration

File: server/src/routes/api/v1/showsRouter.js [hunk 3]
```diff
@@ -21,11 +21,13 @@ showsRouter.get("/:id", async(req,res) => {
...
-    const serializedShow = ShowSerializer.getSummary(show)
+    const serializedShow = await ShowSerializer.getDetail(show)
     return res.status(200).json({ show: serializedShow })
   } catch(err){
     return res.status(500).json({ errors: err })
   }
 })
 
+
+
```
Comment (nwalberts) [inline server/src/routes/api/v1/showsRouter.js]: I dont think we need these extra lines at the bottom

File: server/src/serializers/ReviewSerializer.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class ReviewSerializer {
+  static getSummary(review) {
+    const allowedAttributes = ["reviewBody", "score", "id"]
```
Comment (nwalberts) [inline server/src/serializers/ReviewSerializer.js]: update after updating the migration

File: client/src/components/TVDetailsPage.js [hunk 5]
```diff
@@ -26,11 +27,23 @@ const TVDetailsPage = props => {
     getShow()
   }, [])
 
+  const reviewListItems = show.reviews.map(reviewItem => {
+    return (
+      <ReviewTile
+        key={reviewItem.id}
+        reviewBody = {reviewItem.reviewBody}
+        score = {reviewItem.score}
+      />
```
Comment (nwalberts) [inline client/src/components/TVDetailsPage.js:36]: Convention in react is that there is no space before or after the `=` with these props

File: server/src/models/Review.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      properties: {
+        reviewBody: { type: "string", minLength: 10 },
+        score: { type: "integer" }
```
Comment (nwalberts) [inline server/src/models/Review.js]: score should between 1 and 5

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: server/src/models/Review.js (hunk 6 above)
Comment (JHHalpern) [reply server/src/models/Review.js]: I did a google search, and we can use { type: "integer", minimum: 1, maximum: 5 } for this

Comment (ATrbovic) [reply server/src/models/Review.js]: Okay ill add that in

**Pull request merged**